### PR TITLE
Feature/xray

### PR DESCRIPTION
**Pull request by temyers**
Add AWS Xray support for clients created by awssociate

### CODE REVIEW

**Comment by matt-tyler** [comment]
I think I would prefer to add something that allows someone to pass in the underlying STS instance. 

That way the tracing could be added without needing to explicitly depend it - I've been trying to ensure that this module takes no dependencies (other than peer dependencies) to keep it slim. (otherwise there are serverless bundlers that try to bundle the sdk and bloats the zip file)

I think making the factory constructor take either an STS instance or an STS option would probably be the most reasonable way to do it. Then if we want to to always create an Xray configured client we can write a wrapper where it is being used.

**Comment by temyers** [comment]
But that adds the burden of responsibility for adding XRAY onto the client.

The point of this library is to simplify that process.

We could add XRay as a peer dependency.  But it needs to be bundled with the function (or in a layer) at this time.

**Comment by matt-tyler** [comment]
Well... no, the point of the library was simplify performing role assumption jumps between accounts. I'm concerned with backwards compatibility. 

I'd also be OK with doing a dynamic import of the xray library - if the import fails don't wrap.

I'm also not 100% on what the effect of the AWS node sdk v2 is going to have on Xray - when the other SDK clients have done this it has broken their respective Xray client.

**Comment by temyers** [comment]
In that case, I this API should be focused purely on the role chain assumption, and potentially setting the credentials on a parameter object.

Construction of a AWS SDK service client is outside the SRP of this module.

Passing in the STS constructor makes sense in that case.

File: packages/awssociate/lib/factory.ts [hunk 1]
```diff
@@ -84,6 +85,19 @@ export class ServiceFactory<T, U> implements IWithOptions<T, U>, IWithRoleChain<
     }
 }
 
+/**
+ * Adds AWS XRay decoration to the constructed XRay client, if it has been specified as a dependency.
+ */
+function decorateWithXRayIfAvailable<T>(service: T): T{
+    try{
+        const AWSXRay = require('aws-xray-sdk')
```
Comment (matt-tyler) [inline packages/awssociate/lib/factory.ts]: you probably want to use the dynamic import eg;

```
const xray = await import('aws-xray-sdk')
```
As that will work in browser environments as well as via node. Not that I'm entirely sure what kind of fruit-loop would be doing role-assumption chaining on the client side...

**Pull request merged**